firedrake/slate/static_condensation/hybridization.py

```diff
@@ -538,33 +554,44 @@ def _split_mixed_operator(self):
 
     def _retrieve_options(self, pc):
         get_option = lambda key: PETSc.Options(self.prefix).getString(key, default="")
+        no_valid_options = lambda key_expected_dict: not(all(not get_option(key) or get_option(key) in expected
+                                                             for key, expected in key_expected_dict.items()))
 
         # Get options for Schur complement decomposition
+        if no_valid_options({"ksp_type": ["preonly"], "pc_type": ["fieldsplit"], "fieldsplit_type": ["schur"]}):
+            raise ValueError("The options for the local solve in the hybridization preconditioner are not set correctly.")
```

Comment:
Ah it's weird that it's `pc_fieldsplit_type` in PETSc but the fieldsplit options you set without the pc. But yes, I will change this, I wanted to have the options as PETSc like as possible.

Reply:
Ah I was wondering why you chose a set in `Iterable[tuple[str, set[str]]])` but I guess it's because it will be faster than a list to look up if an element is present.  Yes, will change the check. This will be more useful for the user.

Reply:
I guess the speed is likely almost no difference given one expects only a very few valid values

Reply:
Yes ok. I took your strawman like that in any case, the only thing I changed is that I check for `if value is not default and value not in supported:` so **and** rather than **or**

Reply:
Ah I understand, thanks!

Reply:
Ah the specification of the type only works in python3.9 like that which is why the docs builds fails. I think I would just remove the specification of the type unless you convince me otherwise.

Reply:
I think that's ok, if you want to keep it for older versions it would be:

```
from typing import Iterable, Tuple, Set, AnyStr

Iterable[Tuple[AnyStr, Set[AnyStr]]]
```